app/controllers/foreclosures_controller.rb

```diff
@@ -1,131 +1,133 @@
+
+    @foreclosure = Foreclosure.new({
+      client_id: @id,
+      currently_foreclosed: params[:currently_foreclosed],
+      originating_lender: params[:originating_lender],
+      original_loan_number: params[:original_loan_number],
+      servicer: params[:servicer],
+      servicer_loan_number: params[:servicer_loan_number],
+      monthly_mortgage_payment: params[:monthly_mortgage_payment],
+      loan_term: params[:loan_term],
+      origination_date: params[:origination_date],
+      been_to_court: params[:been_to_court],
+      court_case_number: params[:court_case_number],
+      working_with_lawyer: params[:working_with_lawyer],
+      working_w_agency: params[:working_w_agency],
+      agency: params[:agency]
+      })
+    if @foreclosure.save
+      flash[:success] = ["You've completed the foreclosure application"]
+      redirect_to "/foreclosures/#{@foreclosure.client_id}"
+    else
+      flash[:danger] = @foreclosure.errors.full_messages
+      render :new
+    end
+  end
 
-def show
-  @foreclosure = Foreclosure.find(params[:id])
-end
+  def show
+    @foreclosure = Foreclosure.find(params[:id])
+  end
 
 
-def edit
-  if client_signed_in?
-   @foreclosure = current_client.foreclosure
- elsif user_signed_in?
-  @forelosure = Foreclosure.find(params[:id])
-end
-end
-
-def update
-  if current_client
-    @foreclosure = Foreclosure.find_by(client_id: current_client.id)
-  elsif current_user
-    @foreclosure = Foreclosure.where(client_id: params[:id])
+  def edit
+    if client_signed_in?
+      @foreclosure = current_client.foreclosure
+    elsif user_signed_in?
+      @foreclosure = Foreclosure.find(params[:id])
+    end
   end
 
-  @foreclosure = Foreclosure.where(client_id: current_client.id)[0]
-  if @foreclosure.update({
-    currently_foreclosed: params[:currently_foreclosed],
-    originating_lender: params[:originating_lender],
-    original_loan_number: params[:original_loan_number],
-    servicer: params[:servicer], 
-    servicer_loan_number: params[:servicer_loan_number],
-    monthly_mortgage_payment: params[:monthly_mortgage_payment],
-    loan_term: params[:loan_term], 
-    origination_date: params[:origination_date], 
-    been_to_court: params[:been_to_court], 
-    court_case_number: params[:court_case_number] , 
-    working_with_lawyer: params[:working_with_lawyer],
-    working_w_agency: params[:working_w_agency], 
-    agency: params[:agency]
-    })
-  flash[:success] = "Foreclosure application submitted."
-  redirect_to '/clients/#{@foreclosure.client_id}'
-else
- render :edit
-end
-end
+  def update
+
+    @foreclosure = current_client.foreclosure if current_client
+    @foreclosure = Foreclosure.find(params[:id]) if current_user
+    # binding.pry
+    if @foreclosure.update_attributes(foreclosure_params)
+      flash[:success] = "Foreclosure application submitted."
+      redirect_to "/clients/#{@foreclosure.client_id}"
+    else
+      render :edit
```

Comment:
We can do this in another PR, but maybe we should send a flash message for the sad path as well? Just to let the user know something went wrong?